### PR TITLE
Strip path suffix from pagenum links to fix the posts pagination

### LINKED ISSUE #5861: Path suffix paired URL structure fails with paginate_links()
## Bug Description

This was originally identified by @milindmore22.

When using the path suffix paired URL structure, the paginated links generated via `paginate_links()` results in malformed URLs.

For example, on a non-AMP page, when on blog index, the “older posts” link should go to `/blog/page/2/`. 

On an AMP page, however, when AMP-to-AMP linking is _disabled_, this comes out to be: `/blog/amp/page/2/`.

On the other hand, on an AMP page with AMP-to-AMP linking _enabled_, this comes out to be: `/blog/amp/page/2/amp/`. The last `/amp/` is added by `AMP_Link_Sanitizer`, since it didn't detect that the URL was already pointing to an AMP page.

This is clearly not correct.

We also need to double check that paged content works properly as well as comment pagination links.

## Expected Behaviour

Pagination links generated by `paginate_links()` should result in valid AMP URLs.

## Steps to reproduce

1. Enable Transitional template mode.
2. Enable the path suffix paired URL structure.
3. Navigate to the AMP page for the blog index.
4. Click on "older posts".

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Implementation brief

* 

## QA testing instructions

* See steps to repro

### PR DESCRIPTION
**Pull request by westonruter**
## Summary

Fixes #5861

## Checklist

- [x] Add tests.
- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

**Comment by westonruter** [comment]
@milindmore22 Would you test this to confirm it is fixing the issue you reported?

**Review by milindmore22 — APPROVED** [review]
Tested, confirmed issue has been resolved.

**Comment by westonruter** [comment]
@pierlon Ready for review.

File: tests/php/src/PairedRoutingTest.php [hunk 1]
```diff
@@ -429,15 +428,53 @@ public function test_filter_unique_post_slug( $post_name, $other_existing_post_n
...
+		$this->instance->add_paired_request_hooks();
 		$this->assertEquals( 1000, has_filter( 'old_slug_redirect_url', [ $this->instance, 'maybe_add_paired_endpoint' ] ) );
 		$this->assertEquals( 1000, has_filter( 'redirect_canonical', [ $this->instance, 'maybe_add_paired_endpoint' ] ) );
+		if ( $using_path_suffix ) {
+			$this->assertEquals( 0, has_filter( 'get_pagenum_link', [ $this->instance, 'filter_get_pagenum_link' ] ) );
+		} else {
+			$this->assertFalse( has_filter( 'get_pagenum_link', [ $this->instance, 'filter_get_pagenum_link' ] ) );
+		}
 		$this->assertFalse( has_action( 'wp_head', 'amp_add_amphtml_link' ) );
+		$this->assertEquals( $using_path_suffix, $this->instance->is_using_path_suffix() );
```
Comment (pierlon) [inline tests/php/src/PairedRoutingTest.php]: This check was already done on line 463. Is it necessary once more?

**Review by pierlon — COMMENTED** [review]
This looks good, one quick question though.

File: tests/php/src/PairedRoutingTest.php (hunk 1 above)
Comment (westonruter) [reply tests/php/src/PairedRoutingTest.php]: Ah no. It was redundant. Removed in c467d11c6.

**Review by pierlon — APPROVED** [review]
Nicely done :ok_hand:.

**Pull request merged**